modules/mapper-extras/src/main/java/org/elasticsearch/index/mapper/extras/RankFeatureQueryBuilder.java

```diff
@@ -378,25 +378,25 @@ protected void doXContent(XContentBuilder builder, Params params) throws IOExcep
 
     @Override
     protected Query doToQuery(SearchExecutionContext context) throws IOException {
-        final MappedFieldType ft = context.getFieldType(field);
+        final MappedField mappedField = context.getMappedField(field);
 
-        if (ft instanceof final RankFeatureFieldType fft) {
-            return scoreFunction.toQuery(RankFeatureMetaFieldMapper.NAME, field, fft.positiveScoreImpact());
-        } else if (ft == null) {
+        if (mappedField == null) {
```

Comment:
I reordered the if/else clauses here to move the null check to the top (as we can't otherwise call `mappedField.type()` without a NPE)